server/interceptor/interceptor.go

```diff
@@ -121,9 +126,23 @@ func getRPCInfo(req interface{}, fullMethod string) (*rpcInfo, error) {
 
 	readonly := strings.HasPrefix(methodName, "Get") || strings.HasPrefix(methodName, "List")
 
+	var kind quota.Kind
+	if readonly {
+		kind = quota.Read
+	} else {
+		kind = quota.Write
+	}
+	var specs []quota.Spec
```

Comment:
Does the order of specs matter?  If not, this might be clearer as:

```go
// All RPCs are subject to global and per-user quotas
specs := []quota.Spec{
	{Group: quota.User, Kind: kind, User: user},
	{Group: quota.Global, Kind: kind},
}
if treeID != 0 {
	// Tree-specific RPCs are also subject to per-tree quotas
	specs = append(specs, quota.Spec{
		{Group: quota.Tree, Kind: kind, TreeID: treeID},
	})
}
```